homeassistant/components/light/device_automation.py

```diff
@@ -0,0 +1,42 @@
+"""Provides device automations for lights."""
+import voluptuous as vol
+
+import homeassistant.components.automation.state as state
+from homeassistant.const import CONF_ENTITY_ID, CONF_PLATFORM, CONF_TYPE
+import homeassistant.helpers.config_validation as cv
+from . import DOMAIN
+
+ENTITY_TRIGGERS = [
+    {
+        # Trigger when light is turned on
+        CONF_PLATFORM: '.{}.device_automation'.format(DOMAIN),
```

Comment:
This is to make the generated triggers work without any change of `homeassistant/components/automation/__init__.py`

As an alternative, make it possible to supply a `domain` instead of a `platform`, in which case the resolver https://github.com/home-assistant/home-assistant/blob/e137572ce7487c08bd1312a521c69aeff732251b/homeassistant/components/automation/__init__.py#L408 to instead of:
```python
platform = importlib.import_module('.{}'.format(conf[CONF_PLATFORM]), __name__)
```
do:
```python
domain = importlib.import_module('..{}.device_automation'.format(conf[CONF_DOMAIN]), __name__)
```